ledger-service/http-json/src/main/scala/com/digitalasset/http/WebsocketEndpoints.scala

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2019 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.http
+
+import akka.http.scaladsl.model.HttpMethods._
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.model.ws.{Message, UpgradeToWebSocket}
+import akka.stream.Materializer
+import akka.stream.scaladsl.Flow
+import com.digitalasset.jwt.domain.Jwt
+import com.digitalasset.ledger.api.refinements.{ApiTypes => lar}
+import com.typesafe.scalalogging.StrictLogging
+import scalaz.syntax.std.option._
+import scalaz.{-\/, \/, \/-}
+import scala.concurrent.{ExecutionContext, Future}
+import EndpointsCompanion._
+
+object WebsocketEndpoints {
+  private val tokenPrefix: String = "jwt.token."
+  private val wsProtocol: String = "daml.ws.auth"
+}
+
+class WebsocketEndpoints(ledgerId: lar.LedgerId,
+                         decodeJwt: ValidateJwt,
+                         webSocketService: WebSocketService)
+                        (implicit mat: Materializer, ec: ExecutionContext) extends StrictLogging {
+
+  import WebsocketEndpoints._
+
+  lazy val transactionWebSocket: PartialFunction[HttpRequest, Future[HttpResponse]] = {
+    case req@HttpRequest(GET, Uri.Path("/transaction/connect"), _, _, _) =>
```

Comment:
I would call this endpoint `/transactions`, `connect` does not add anything to it. Unless it is a common practice to call websocket endpoints this way.
cc @bame-da

Reply:
yeah, I added this trying to follow `contracts/search`, `command/create` convention and imply it's not a REST endpoint by `/connect` . On the other hand client may know from `ws://` so I can remove `/connect` if more people think it's useless